lib/teleagent/agent.go

```diff
@@ -85,9 +85,10 @@ func (a *AgentServer) Serve() error {
 				return trace.Wrap(err, "unknown error")
 			}
 			if !neterr.Temporary() {
-				if !strings.Contains(neterr.Error(), "use of closed network connection") {
-					log.Errorf("got permanent error: %v", err)
+				if strings.Contains(neterr.Error(), "use of closed network connection") {
+					return nil
 				}
+				log.Errorf("got permanent error: %v", err)
 				return err
```

Comment:
`trace.Wrap(err)`?

Reply:
@andrejtokarcik Sorry, saw you resolve this - so you don't think `trace.Wrap` is needed here?

Reply:
Oh sorry, good catch, I totally misunderstood your message here. I thought you were referring to my changed line, i.e. suggesting to wrap the `err` being passed to the log function - which sounded a little strange but I thought I resolved it in the other way. Sure, I'll add `trace.Wrap` to the return statement (although it wasn't originally present - probably just a mishap).